server/src/main/java/io/spine/server/delivery/FailedReception.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2022, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.delivery;
+
+import io.spine.annotation.Internal;
+import io.spine.annotation.SPI;
+import io.spine.base.Error;
+
+/**
+ * The evidence of an {@link InboxMessage} which has failed to be handled
+ * by its receptor, such as an event or a command handler method.
+ *
+ * <p>End-users may choose to do one of the following:
+ *
+ * <ul>
+ *     <li>{@linkplain #markDelivered() mark} the message as delivered;
+ *     <li>{@linkplain #repeatDispatching() repeat dispatching} of the message immediately.
+ * </ul>
+ *
+ * <p>Alternatively, end-users may choose to define their own way of reacting
+ * to reception failures by implementing a custom {@code FailedReception.Action},
+ * and returning it via the corresponding
+ * {@linkplain DeliveryMonitor#onReceptionFailure(FailedReception) API} of {@code DeliveryMonitor}.
+ */
+public final class FailedReception {
```

Comment:
Don't we need `hashCode()`, `equals()`, and `toString()` in this class?

Reply:
@alexander-yevsyukov not sure why we would need those. Let's discuss vocally.